src/NLitecoin/Litecoin.fs

```
@@ -0,0 +1,358 @@
+            .SetBase58Bytes(Base58Type.EXT_SECRET_KEY, [| 0x04uy; 0x88uy; 0xADuy; 0xE4uy |])
+            .SetNetworkStringParser(LitecoinMainnetAddressStringParser())
+            .SetBech32(Bech32Type.WITNESS_PUBKEY_ADDRESS, bech32)
+            .SetBech32(Bech32Type.WITNESS_SCRIPT_ADDRESS, bech32)
+            .SetBech32(Bech32Type.TAPROOT_ADDRESS, bech32)
+            .SetMagic(0xdbb6c0fbu)
+            .SetPort(9333)
+            .SetRPCPort(9332)
+            .SetName("ltc-main")
+            .AddAlias("ltc-mainnet")
+            .AddAlias("litecoin-mainnet")
+            .AddAlias("litecoin-main")
+            .SetUriScheme("litecoin")
+            .AddDNSSeeds(
+                [|
+                    DNSSeedData("loshan.co.uk", "seed-a.litecoin.loshan.co.uk")
+                    DNSSeedData("thrasher.io", "dnsseed.thrasher.io")
+                    DNSSeedData("litecointools.com", "dnsseed.litecointools.com")
+                    DNSSeedData("litecoinpool.org", "dnsseed.litecoinpool.org")
+                    DNSSeedData("koin-project.com", "dnsseed.koin-project.com")
+                |])
+            .AddSeeds(NetworkSetBase.ToSeed pnSeed6_main)
+            .SetGenesis("010000000000000000000000000000000000000000000000000000000000000000000000d9ced4ed1130f7b7faad9be25323ffafa33232a17c3edf6cfd97bee6bafbdd97b9aa8e4ef0ff0f1ecd513f7c0101000000010000000000000000000000000000000000000000000000000000000000000000ffffffff4804ffff001d0104404e592054696d65732030352f4f63742f32303131205374657665204a6f62732c204170706c65e280997320566973696f6e6172792c2044696573206174203536ffffffff0100f2052a010000004341040184710fa689ad5023690c80f3a49c8f13f8d45b8c857fbcbc8bc4a8e4d3eb4b10f4d4604fa08dce601aaf0f470216fe1b51850b4acf21b179c45070ac7b03a9ac00000000")
+    
+    /// not used in GWallet
+    override self.CreateTestnet() =
+        null
+
+    /// not used in GWallet
+    override self.CreateRegtest() =
+        null
```

@webwarrior-ws better FAIL FAST, which means throw, not null (do this in other parts of the code please)

---

We can't throw an exception here. `NetworkSetBase`, `Litecoin`'s superclass, will try to initialize all three instances, even if only one is used. Our only option is to return null.

---

didn't we agree that it doesn't need to inherit from NetworkBase?

---

No, it needs to inherit from `NetworkBase`.
See https://github.com/MetacoSA/NBitcoin/blob/93ef4532b9f2ea52b2c910266eeb6684f3bd25de/NBitcoin/NetworkSet.cs#L69C12-L69C12

---

@webwarrior-ws but is NetworkBase used in geewallet? I believe we already had this conversation and you said that it is not

---

Not directly. But it is used by `NetworkBuilder` to build `Network`, and that **is** used in GWallet.

---

ok fair enough; I'm then marking this convo as resolved with the `Resolve conversation` button